src/crystal/compiler_rt/shift.cr

```diff
@@ -0,0 +1,53 @@
+# :nodoc:
+# Ported from https://github.com/llvm/llvm-project/blob/ce59ccd04023cab3a837da14079ca2dcbfebb70c/compiler-rt/lib/builtins/ashlti3.c
+#
+# Returns: a << b
+# Precondition:  0 <= b < bits_in_tword
+fun __ashlti3(a : Int128, b : Int32) : Int128
+  low, high = a.unsafe_as(Tuple(UInt64, Int64))
+  if b >= 64
+    low, high = 0u64, (low << (b &- 64)).to_i64!
```

Comment:
It seems a bit inconsistent to use `&-` but not `unsafe_shl` and `unsafe_shr`. Is it intentional to check for overflow for the shift operations?

Reply:
The original c source also just used `<<` without any checking, so it should be fine to replace it. That being said, most of crystal's compiler-rt uses `<<` instead of `unsafe_shl`.. probably because it feels very natural to replace for ex. `+` with `&+` but that is not the case for `<<` and `unsafe_shl`

Reply:
I just didn't think about using `unsafe_shl` for shifting, and `&+` was indeed natural. Should I change it?

Reply:
If you feel like it. Perhaps more worth it to focus on feature completeness than on performance like this at this stage?

Reply:
Done